packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -227,20 +226,16 @@ export function getValidatorApi({
       // the first `MAXIMUM_GOSSIP_CLOCK_DISPARITY` duration of the epoch `tolerantCurrentEpoch`
       // will equal `currentEpoch + 1`
 
-      const duties: routes.validator.AttesterDuty[] = [];
-      for (const validatorIndex of validatorIndices) {
-        const validator = state.validators[validatorIndex];
-        if (!validator) {
-          throw new ApiError(400, `Validator index ${validatorIndex} not in state`);
-        }
-        const duty = assembleAttesterDuty(
-          config,
-          {pubkey: validator.pubkey, index: validatorIndex},
-          state.epochCtx,
-          epoch
-        );
-        if (duty) duties.push(duty);
-      }
+      const duties = state.epochCtx.getCommitteeAssignments(
+        epoch,
+        state.validators.map((validator, k) => {
```

Comment:
I suppose we need to map `validatorIndices` instead of `state.validators`

Reply:
I use state.validators because we need the pubkey as well.

Reply:
Running `state.validators.map` must be a very slow operation, does that show up in the benchmarks? If you need the pubkey use the index2pubkey cache

Reply:
@dapplion when I try using index2pubkey, it seems it's actually slower than using state.validators (maybe because I have to convert using `toBytes()`.  here's the benchmark (I updated the benchmark test to have all 3 of these run calls in there too):

```
get attester duties
================================================================
new way: getCommitteeAssignments                                       56.21777 ops/s      17.78797 ms/op      6 runs   0.1070 s
new way (plus index2pubkey): getCommitteeAssignments                   14.70149 ops/s      68.02031 ms/op      2 runs   0.1360 s
old way: assembleAttesterDuty batch                                   0.1696719 ops/s      5.893728  s/op      1 runs    11.79 s
```

Reply:
I think @tuyennhv  means:
```ts
const validators = state.validators.persistent;
const duties = state.epochCtx.getCommitteeAssignments(
  epoch,
  validatorIndices.map((index) => {
    const validator = validators.get(index);
    if (!validator) {
      throw new ApiError(400, `Validator index ${index} not in state`);
    }
    return {pubkey: validator.pubkey, index};
  })
);
```

Reply:
As it stands, this is just passing _all_ validators' pubkey + a different index to `getCommitteeAssignments`? Doesn't seem right.

Reply:
ok, made a few changes and now the benchmarks are as follows (which make a lot more sense than the original ones i posted):

```
get attester duties
================================================================
new way: getCommitteeAssignments                                      0.6062110 ops/s      1.649591  s/op      2 runs    3.299 s
new way (plus index2pubkey): getCommitteeAssignments                  0.7766266 ops/s      1.287620  s/op      2 runs    2.576 s
old way: assembleAttesterDuty batch                                   0.2985871 ops/s      3.349106  s/op      2 runs    6.698 s
    ✓ performance comparison (12678ms)
```

so i'm using `index2pubkey` now